### PR TITLE
For payments initiated by payment request buttons, set form top error container

### LINKED ISSUE #879: Intent verification errors not properly displayed after Payment Request
When checking out via Payment Request Button from the Checkout screen (available using a filter [[doc](https://docs.woocommerce.com/document/stripe/#section-23)]) using a 3D Secure card, and when the verification fails, an error shows up in the payment method section below (below the form or a saved card).

Edit. i.e. here:

Screen Shot 2020-12-07 at 3 03 56 PM

From the Cart or Single Product screen, the **error doesn't show at all** (other than in the JS console).

Tested with Chrome using card `4000000000003063` in test mode.

Note – before PaymentIntents, the flow would redirect back to Checkout with an error on top:

Screen Shot 2019-05-30 at 7 18 43 PM

### PR DESCRIPTION
**Pull request by allendav**
# Changes proposed in this Pull Request:

Issue: Link to the GitHub issue this PR addresses (if appropriate).
Fixes #879 

Description: If a payment request button is used to initiate payment and fails (e.g. using a 3D Secure test card and then clicking FAIL when prompted) ensure the error is displayed to the user (and not just console logged.) To pull this off, we set a data flag on the payment request button itself when it is used to initiate payment. If an error occurs and that flag is present, it prompts us to use the woocommerce-notices-wrapper div as the error container.

Images or gifs: Include before and after screenshots or gifs/videos when it makes sense.

Screen Shot 2020-12-07 at 3 53 08 PM
Screen Shot 2020-12-07 at 3 52 32 PM
Screen Shot 2020-12-07 at 3 51 54 PM

# Testing instructions

- Install this branch and activate it
- Ensure you are set up with keys and test mode for the Stripe plugin and that payment request buttons are enabled
- Add this filter:

`add_filter( 'wc_stripe_show_payment_request_on_checkout', '__return_true' );`

- Navigate to a product page using Chrome
- Click on the payment request button. When prompted, enter the 4000 0000 0000 3063 test card
- Enter a dummy CCV when prompted (e.g. 123)
- Wait for the Authorize/Fail modal. Click Fail.
- Ensure the error message is displayed
- Repeat for cart and checkout pages

-------------------
- [x] Make sure your changes respect [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- [ ] Did you make changes, or create a **new .js file**? If **Gruntfile.js** exists in the repo, make sure to run `grunt`.

(remove me)
- [x] Did I add a title? A descriptive, yet concise, title.
- [ ] How can this code break?
- [ ] What are we doing to make sure this code doesn't break?

### CODE REVIEW

**Review by reykjalin — CHANGES_REQUESTED** [review]
I'm happy with the general approach here, but this doesn't seem to be working for me.

Using the default gateway configuration for the payment request buttons and Brave nothing happens.

I think this is because the code where we [set the initiator data property](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1402/files#diff-73c011ce1388f1432621c0fb207e47201e82f70fdf7e4a9b0d44eb47a9cb515bR582-R586) is never entered with the payment request button type set to `Buy`, as seems to be the default:

image

This is because the check used for the payment request buttons is just checking for `Custom` or `Branded`.

I assume this is supposed to work for any type of payment request button, so we probably need to address this?

I think we just set the `data` property on the button, regardless of the type of button 🤔

**Comment by allendav** [comment]
@reykjalin - I didn't think to test with it set to other settings - I'll fix this asap

**Comment by allendav** [comment]
@reykjalin - in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1402/commits/c48bd3a7ea7b847af8a87b721325f11f1580b04e I changed a few things. We first bind a click handler for any button type and any page. That should take care of what you're seeing. Be sure to have Disable cache (while Dev Tools is open) so you don't get old cached JS from before. I also chose to simplify the flagging. It applies a class (not a data) to the body element.

Please let me know if you have any concerns or problems with it and thanks again for your thorough testing!!!!11!!

File: assets/js/stripe-payment-request.js [hunk 1]
```diff
@@ -548,6 +548,12 @@ jQuery( function( $ ) {
 		},
 
 		attachPaymentRequestButtonEventListeners: function( prButton, paymentRequest ) {
+			// First, attach a flag setting event so we can know when a PRB has initiated payment
+			prButton.on( 'click', function ( evt ) {
+				$( 'body' ).addClass( "woocommerce-stripe-prb-clicked" );
+			});
```
Comment (reykjalin) [inline assets/js/stripe-payment-request.js:554]: Similarly to my other comment, we should probably include _why_ we need to know when a PRB initiated a payment 🙂

**Review by reykjalin — CHANGES_REQUESTED** [review]
Works like a charm! 🥇 

I think we should just add some comments explaining why we're doing this, but otherwise we're ready to merge 👏

**Comment by reykjalin** [comment]
@allendav I'm labeling this as `Ready to Merge` despite requesting changes to reflect that I don't think we need another review+approval before we merge once the comments have been addressed :)

Also, the branch needs a rebase on `master` before being merged :)

**Comment by allendav** [comment]
Updated comments and rebased

**Pull request merged**